src/subscription_manager/gui/firstboot/rhsm_login.py

```diff
@@ -151,15 +152,19 @@ def focus(self):
         # login_text.grab_focus()
 
     def initializeUI(self):
-        log.debug("initializeUi %s", self)
         # Need to make sure that each time the UI is initialized we reset back
         # to the main register screen.
 
+        if self.finished:
+            self.register_widget.done()
+            return
+
         # Note, even if we are standalone firstboot mode (no rhn modules),
         # we may still have RHN installed, and possibly configured.
         self._read_rhn_proxy_settings()
-
         self.register_widget.initialize()
+        # Make sure to show the unregister screen
+        self.register_widget.info.set_property('do-unregister', True)
```

Comment:
(for later). If only FirstbootWidget cares about this, and we don't need notify signals, this could
just be a class attribute.